W-18359536 OpenTelemetry / Distributed Tracing (#894)
mulesoft/docs-gateway: gateway/1.10/modules/ROOT/pages/flex-local-tracing-config.adoc

= Configuring Flex Gateway Distributed Tracing in Local Mode
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../assets/images
:page-mode: local

include::partial$flex-otel-support.adoc[]

== Enable Distributed Tracing with a YAML Configuration Resource

Edit and then apply this configuration resource to enable distributed tracing. This configuration resource applies to all API instances.

> This example sets sampling to 0 for all traces, and sampling will be configured on a per-API basis using policies.

Reviewer: ```suggestion
This example sets sampling to 0 for all traces, and sampling is configured on a per-API basis using policies.
```

[source,yaml]
----
apiVersion: gateway.mulesoft.com/v1beta1
kind: Configuration
metadata:
  name: tracing
  namespace: test-ns
spec:
  tracing:
    provider:
      type: anypoint
      properties:
        timeout: 10s
    sampling:
      client: 0   # no sampling - for the case where you want to configure sampling on a per-API basis using the Tracing policy.
      random: 0   # no sampling - for the case where you want to configure sampling on a per-API basis using the Tracing policy.
      overall: 0  # no sampling - for the case where you want to configure sampling on a per-API basis using the Tracing policy.

>       # # Add labels such as these example labels,if you want to configure labels globally. You can also configure labels on a per-API basis using the Tracing policy.

Reviewer: ```suggestion
      # # Add labels such as these example labels, if you want to configure labels globally. You can also configure labels on a per-API basis by using the Tracing policy.
```
